tests/subsys/bootloader/b0_lock_rwx/src/main.c

```diff
@@ -43,14 +60,18 @@ void *get_config(void)
 			(NRF_RRAMC_REGION_PERM_WRITE_MASK),
 			"Write permission isn't cleared");
 #endif
-	zassert_true(config.size_kb > 0, "Protected region has zero size.");
+	zassert_equal(config.size_kb, RRAMC_REGION_FOR_TEST_SIZE / 1024,
+		      "Protected region size doesn't match protected partition size.");
+	zassert_equal(
+		config.address, RRAM_REGION_FOR_TEST_ADDRESS,
+		"Protected region start address doesn't match protected partition start address.");
 	return NULL;
 }
 
-ZTEST(b0_self_lock_test, test_reading_b0_image)
+ZTEST(b0_self_lock_test, test_rwx_locked_region)
 {
 	printk("Region %d\n", RRAMC_REGION_FOR_TEST);
-	uint32_t protected_end_address = 1024 * config.size_kb;
+	uint32_t protected_end_address = config.address + (1024 * config.size_kb);
 	volatile uint32_t *unprotected_word = (volatile uint32_t *)protected_end_address;
 	volatile uint32_t *protected_word =
 		(volatile uint32_t *)protected_end_address - sizeof(uint32_t);
```

Comment:
additional check? something like that:

uint32_t tmp = nrf_rramc_region_config_raw_get(NRF_RRAMC, RRAMC_REGION_FOR_TEST);
.
.
.
zassert_equal(nrf_rramc_region_config_raw_get(NRF_RRAMC, RRAMC_REGION_FOR_TEST) !=
            tmp, "error, managed to change permissions")